### DISCUSSION TITLE
getClusterLeaves doesn't work on Android

### DISCUSSION THREAD

**Discussion by Plat1** [rnmapbox/maps#1534] [Q&A] [answered]
Hi,
I'm having a big problem with the getClusterLeaves async function.
Everything works fine with iOs but there's no way to make it work with Android,
I can not have a response. 
Here is a code, I can not have data to populate "collection" only when on Android device,
has anyone been able to use getClusterLeaves?
getClusterLeaves on Android appears to work ONLY if I put fix integers and not variables,
like this: const collection = await this._cluster.getClusterLeaves(3, 4, 0);
... but this would be nosense.

Thank u very much to all!
Elena

<MapboxGL.ShapeSource
              cluster={true}
              clusterRadius={50}
              clusterMaxZoom={14}
              id="earthquakes"
              shape={this.state.json}
              ref={(c) => (this._cluster = c)}
              onPress={async (e) => {

                if (e.features[0].properties["cluster"]) {
                    let cluster_id = e.features[0].id;
                    let limit = e.features[0].properties.point_count;

                    const collection = await this._cluster.getClusterLeaves(cluster_id, limit, 0);   // PROBLEM HERE

                    let visibleAlbums = [];
                    for (let i = 0; i < limit; i++) {
                      visibleAlbums.push(collection.features[i].properties);
                    }
                    this.navigation.navigate('Advanced search', { albums: visibleAlbums });
                    return;
                  }
                }
                e.features[0].properties["from"] = "explore";
                global.Explore2DScreenRef.navigation.navigate("Album", e.features[0].properties);
                this.state.results[1] = e.features[0].geometry.coordinates[1];
                this.state.results[0] = e.features[0].geometry.coordinates[0];
              }}

            >

**Comment by ferdicus** [comment]
hi, which version of the lib are you on? 
Also, I don't really understand what your issue is. 
Does it work or not? Or does it work, but only in a way you do not like? 
You say it's working on iOS, how did you get it to work there? Can you share some code please?

**Comment by dantereve** [comment, marked as answer]
Hi ! @Plat1, I was facing the same issue. 

The id in your `let cluster_id = e.features[0].id;` is a string. You should use `let cluster_id = e.features[0].properties.cluster_id` instead.  Now you'll get a number as required by the method `.getClusterLeaves()`. 


Here is anexemple of a cluster's Feature.
``` json
{
    "geometry": {
        "coordinates": [
            0.0589439868927,
            0.0384530463661
        ],
        "type": "Point"
    },
    "id": "83",
    "properties": {
        "cluster": true,
        "cluster_id": 83,
        "point_count": 2,
        "point_count_abbreviated": "2"
    },
    "type": "Feature"
}
```

**Reply by Plat1** [reply]
Fantastic! @dantereve you save me a lot of troubles! 
this was the only code to modify
let cluster_id = e.features[0].properties.cluster_id  
... and voilà!
I still don't know why iOs could process it anyway, but now everything is fine on Android too.
Thank you very much.
Elena
